airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -229,9 +229,9 @@ def parse_response(
     ) -> Iterable[Mapping]:
         r = response.json()
 
-        dimensions = [h.get("name").replace(":", "_") for h in r.get("dimensionHeaders", [{}])]
-        metrics = [h.get("name").replace(":", "_") for h in r.get("metricHeaders", [{}])]
-        metrics_type_map = {h.get("name").replace(":", "_"): h.get("type") for h in r.get("metricHeaders", [{}])}
+        dimensions = [h.get("name").replace(":", "_") if "name" in h else None for h in r.get("dimensionHeaders", [{}])]
+        metrics = [h.get("name").replace(":", "_") if "name" in h else None for h in r.get("metricHeaders", [{}])]
+        metrics_type_map = {h.get("name").replace(":", "_"): h.get("type") for h in r.get("metricHeaders", [{}]) if "name" in h}
```

Comment:
the old behavior would actually end up with a dictionary that overwrites the None key for any metrics without a name. To me this feels a bit nonsensical so I opted to just skip over metrics missing the name.

As mentioned in the description, I would also be okay if we did the same for the other statements. It's much cleaner, but I'm not sure if there were downstream affects if we deviated from what we considered the earlier working behavior.